internal/sops/gcpkms/keysource_test.go

```diff
@@ -0,0 +1,165 @@
+package gcpkms
+
```

Comment:
There are missing tests for some of the methods, in addition, it needs `_Compat` tests like https://github.com/fluxcd/kustomize-controller/blob/main/internal/sops/hcvault/keysource_test.go#L132, https://github.com/fluxcd/kustomize-controller/blob/main/internal/sops/hcvault/keysource_test.go#L204 and https://github.com/fluxcd/kustomize-controller/blob/main/internal/sops/hcvault/keysource_test.go#L224 to ensure we maintain compatibility with SOPS itself.

Reply:
I think it would need to be an integration test, like in Azure since SOPS will call out to Google's API